File: frameworks/config.center/src/main/java/com/example/scaffold/frameworks/config/center/ArgumentResolverConfigurer.java
package com.example.scaffold.frameworks.config.center;

import org.springframework.context.annotation.Configuration;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

@Configuration
public class ArgumentResolverConfigurer implements WebMvcConfigurer {
    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
        resolvers.add(new PayloadHandlerMethodArgumentResolver(List.of(new MappingJackson2HttpMessageConverter())));
    }
}

File: frameworks/bean.core/src/main/kotlin/com/qomolangma/frameworks/bean/core/ArgumentResolverConfigurer.kt
package com.qomolangma.frameworks.bean.core

import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class ArgumentResolverConfigurer : WebMvcConfigurer {
    override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
        resolvers.add(PayloadHandlerMethodArgumentResolver(listOf(MappingJackson2HttpMessageConverter())))
    }
}
